### PR TITLE
Istio Ingress

### PR DESCRIPTION
**Pull request by aj-jester**
Live URL: http://demo-aj.somebigbank.com/

```
aj@demo-aj-somebigbank-com:~$ kubectl get gateway
NAME              AGE
httpbin-gateway   8m9s
aj@demo-aj-somebigbank-com:~$ kubectl get virtualservice
NAME      GATEWAYS            HOSTS   AGE
httpbin   [httpbin-gateway]   [*]     8m14s
aj@demo-aj-somebigbank-com:~$ kubectl get service | grep -i httpbin
httpbin              ClusterIP   10.96.1.205   <none>        8000/TCP        8m24s
aj@demo-aj-somebigbank-com:~$ kubectl get po | grep -i httpbin
httpbin-779c54bf49-g7kcj              1/1     Running     0          8m30s
```

Screen Shot 2020-06-08 at 10 09 40 AM

### CODE REVIEW

File: istio/manifests/config.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        - name: admin_password
+          title: Admin Password
+          type: password
+          default: 'admin'
+          required: true
+        - name: grafana_replicas
+          title: Replicas
+          type: text
+          required: true
+    - name: ingress
```
Comment (dexhorthy) [inline istio/manifests/config.yaml]: are any of these config options used?

Comment (aj-jester) [reply istio/manifests/config.yaml]: One of them, removed the rest.

File: istio/manifests/preflight.yaml [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+              message: Requires at Kubernetes 1.13.0 or later, and recommends 1.15.0.
+              uri: https://www.kubernetes.io
+          - warn:
+              when: "< 1.15.0"
+              message: Your cluster meets the minimum version of Kubernetes, but we recommend you update to 1.15.0 or later.
+              uri: https://kubernetes.io
+          - pass:
+              when: ">= 1.15.0"
+              message: Your cluster meets the recommended and required versions of Kubernetes.
+    - customResourceDefinition:
```
Comment (dexhorthy) [inline istio/manifests/preflight.yaml]: Is this still a required check @aj-jester ? Can we maybe remove any checks that aren't relevant to this app?

Comment (aj-jester) [reply istio/manifests/preflight.yaml]: Removed some of the ones that aren't relevant.

File: istio/manifests/config.yaml [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+spec:
+  groups:
+    - name: istio
+      title: Istio
+      description: Istio Configuration
+      items:
+        - name: ingress_host
+          title: Ingress Host
+          type: text
+          default: "*"
```
Comment (dexhorthy) [inline istio/manifests/config.yaml:14]: 👍

**Review by dexhorthy — APPROVED** [review]

File: istio/README.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+## Make a release
+
+To install kURL without Contour Ingress
+
+* Run `make release-kurl-installer`.
+
+To install kURL with Istio along with a test app.
+
+* Run `make release`
```
Comment (austinchambers) [inline istio/README.md:11]: I'd think the README.md would be primarily about installing this kotsapp. So to have the `kots install` and kurl commands here (one for existing cluster, one for embedded clusters). Also with the associated License available for use. 

The `make release` commands seem to be more oriented around contributing to the kotsapp itself.

File: istio/README.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+## Sample App
+
+By default there is a sample app listening on port 80 included as well to demonstrate some basic capabilities for Istio.
+
+Once Istio is deployed go to `http://<VM_IP_address>` to view the sample app.
+
+## Istio TLS
+
+TODO
```
Comment (austinchambers) [inline istio/README.md:21]: Super curious about TLS

**Review by austinchambers — APPROVED** [review]

**Pull request merged**